core/sdk-core/src/test/java/software/amazon/awssdk/core/SdkNumberTest.java

```diff
@@ -207,7 +207,8 @@ private void assertEqualitySDKNumberWithNumber(SdkNumber sdkNan, Number nanDoubl
         assertThat(sdkNan.intValue()).isEqualTo(nanDouble.intValue());
         assertThat(sdkNan).hasToString(nanDouble.toString());
         assertThat(sdkNan.byteValue()).isEqualTo(nanDouble.byteValue());
-        assertThat(sdkNan.doubleValue()).isEqualTo(nanDouble.doubleValue());
+        // convert expected to nullable Double to prevent comparison of primitives because Double.NaN == Double.NaN is false
+        assertThat(sdkNan.doubleValue()).isEqualTo(Double.valueOf(nanDouble.doubleValue()));
```

Comment:
Should we convert the left side as well, just to be explicit that we're not comparing primitives?

Reply:
Sure sounds good